### DISCUSSION TITLE
[Support]求教，Swagger生成的uri如何避免以斜杠结尾？

### DISCUSSION THREAD

**Discussion by googoo625** [midwayjs/midway#3943] [Q&A]
### Describe the problem（描述问题）

~~~js
@Controller("/groups" )
export class GroupController {
	@Get()
	public async getGroups(@Query("parent") parent: number): Promise<ROGroupArray> {
		//
	}
	
	@Post()
	public async create(@Body() data: Group): Promise<void> {
		//
	}
	...
}
~~~

如上，swagger提取的几个接口uri都是"/groups/"，如何避免接口以斜杠结尾？

### Midway Versions（Midway 版本）

npx midway-version@latest
**************************************************
>> Current version @midwayjs/decorator(3.16.0) and @midwayjs/core(3.16.2) not found in @midwayjs/version, please check it.
**************************************************

执行命令如上。package.json中的配置是：
   "@midwayjs/bootstrap": "^3.16.0",
    "@midwayjs/core": "^3.16.2",
    "@midwayjs/decorator": "^3.16.0",
    "@midwayjs/koa": "^3.16.1",
    "@midwayjs/logger": "^3.4.0",
    "@midwayjs/swagger": "^3.16.1",

**Comment by czy88840616** [comment]
避免不了吧，除非 @Controller("/" ) ，路由里写 /groups

**Reply by googoo625** [reply]
目前也只有这样处理了。
建议@midwayjs/swagger改进一下，REST规范不建议uri以斜杠结尾，API规范检查时会警告。
Java版本的就没这问题，不会以斜杠结尾。

**Comment by waitingsong** [comment]
rest 只是个规范并非标准。没必要花时间改代码仅仅为了符合它。

**Reply by czy88840616** [reply]
这个不是swagger的问题，路由是koa-router的分组创建的，在 new Router（“groups”）之后至少得有个/

**Reply by waitingsong** [reply]
对嘛， rest 自己搞个所谓标准，然而软件（koa-router）不支持它。电工查漏电一路修到发电厂去了～

**Comment by googoo625** [comment]
目前无法解决，只能修改代码，把@Controller里的前缀改到每个接口上声明。
